systems/plants/RigidBody.h

```diff
@@ -44,23 +44,31 @@ class DLLEXPORT_RBM RigidBody {
   int robotnum; // uses 0-index. starts from 0
   static const std::set<int> defaultRobotNumSet;
 // note: it's very ugly, but parent,dofnum,and pitch also exist currently (independently) at the rigidbodymanipulator level to represent the featherstone structure.  this version is for the kinematics.
-  int parent;
+  std::shared_ptr<RigidBody> parent;
+  int body_index; // index in RBM bodies vector (set in compile()) TODO: remove after we finish the conversion from parents being body indices to being pointers
```

Comment:
this was the big change that forced a number of small changes throughout the code.  body_index makes it easy to get the old information, but this version makes it possible to reorder bodies, etc, in RBM::compile()